crates/common/fork_choice/src/handlers.rs

```diff
@@ -0,0 +1,210 @@
+    execution_engine: &impl ExecutionApi,
+) -> anyhow::Result<bool> {
+    // `retrieve_blobs_and_proofs` is implementation and context dependent
+    // It returns all the blobs for the given block root, and raises an exception if not available
+    // Note: the p2p network does not guarantee sidecar retrieval outside of
+    // `MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS`
+
+    let Some(beacon_block) = store.blocks.get(&beacon_block_root) else {
+        bail!("could not get beack block");
+    };
+    let blob_versioned_hashes = blob_versioned_hashes(&beacon_block.body.execution_payload)?;
+    let blobs_and_proofs = execution_engine
+        .engine_get_blobs_v1(blob_versioned_hashes)
+        .await?;
+
+    let mut blobs = vec![];
+    let mut proofs = vec![];
+
+    for block_and_proof in blobs_and_proofs {
+        let block_and_proof =
+            block_and_proof.ok_or_else(|| anyhow::anyhow!("Invalid proposer index"))?;
+        blobs.push(block_and_proof.blob);
+        proofs.push(block_and_proof.proof);
+    }
+
+    verify_blob_kzg_proof_batch(&blobs, blob_kzg_commitments, &proofs)?;
+    Ok(true)
+}
+
+pub fn get_ancestor(store: &Store, root: B256, slot: u64) -> B256 {
+    if let Some(block) = store.blocks.get(&root) {
+        if block.slot > slot {
+            return get_ancestor(store, block.parent_root, slot);
+        }
+    }
+    root
+}
+
+/// Compute the checkpoint block for epoch ``epoch`` in the chain of block ``root``
+pub fn get_checkpoint_block(store: &Store, root: B256, epoch: u64) -> B256 {
+    let epoch_first_slot = compute_start_slot_at_epoch(epoch);
+    get_ancestor(store, root, epoch_first_slot)
+}
+
+/// Update checkpoints in store if necessary
+pub fn update_checkpoints(
+    store: &mut Store,
+    justified_checkpoint: Checkpoint,
+    finalized_checkpoint: Checkpoint,
+) -> anyhow::Result<()> {
+    // Update justified checkpoint
+    if justified_checkpoint.epoch > store.justified_checkpoint.epoch {
+        store.justified_checkpoint = justified_checkpoint
+    }
+
+    // Update finalized checkpoint
+    if finalized_checkpoint.epoch > store.finalized_checkpoint.epoch {
+        store.finalized_checkpoint = finalized_checkpoint
+    }
+    Ok(())
+}
+
+/// Update unrealized checkpoints in store if necessary
+pub fn update_unrealized_checkpoints(
+    store: &mut Store,
+    unrealized_justified_checkpoint: Checkpoint,
+    unrealized_finalized_checkpoint: Checkpoint,
+) -> anyhow::Result<()> {
+    // Update unrealized justified checkpoint
+    if unrealized_justified_checkpoint.epoch > store.unrealized_justified_checkpoint.epoch {
+        store.unrealized_justified_checkpoint = unrealized_justified_checkpoint
+    }
+
+    // Update unrealized finalized checkpoint
+    if unrealized_finalized_checkpoint.epoch > store.unrealized_finalized_checkpoint.epoch {
+        store.unrealized_finalized_checkpoint = unrealized_finalized_checkpoint
+    }
+
+    Ok(())
+}
```

Comment:
```suggestion
/// Update unrealized checkpoints in store if necessary
pub fn update_unrealized_checkpoints(
    store: &mut Store,
    unrealized_justified_checkpoint: Checkpoint,
    unrealized_finalized_checkpoint: Checkpoint,
) {
    // Update unrealized justified checkpoint
    if unrealized_justified_checkpoint.epoch > store.unrealized_justified_checkpoint.epoch {
        store.unrealized_justified_checkpoint = unrealized_justified_checkpoint
    }

    // Update unrealized finalized checkpoint
    if unrealized_finalized_checkpoint.epoch > store.unrealized_finalized_checkpoint.epoch {
        store.unrealized_finalized_checkpoint = unrealized_finalized_checkpoint
    }
}
```

same reason as above